### PR TITLE
renaming the envoyproxy & gateway provider structures.

### PR DESCRIPTION
**Pull request by qicz**
support  [https://github.com/envoyproxy/gateway/pull/1218#issuecomment-1491167762](https://github.com/envoyproxy/gateway/pull/1218#issuecomment-1491167762)

### CODE REVIEW

File: api/config/v1alpha1/envoygateway_types.go [hunk 1]
```diff
@@ -68,9 +68,9 @@ type Gateway struct {
 	ControllerName string `json:"controllerName,omitempty"`
 }
 
-// Provider defines the desired configuration of a provider.
+// GatewayProvider defines the desired configuration of a provider.
 // +union
-type Provider struct {
+type GatewayProvider struct {
```
Comment (arkodg) [inline api/config/v1alpha1/envoygateway_types.go:73]: should this be `EnvoyGatewayProvider` ? since the data plane provider is called `EnvoyProxy`

Comment (qicz) [reply api/config/v1alpha1/envoygateway_types.go]: done @arkodg

File: api/config/v1alpha1/envoygateway_types.go [hunk 2]
```diff
@@ -80,23 +80,23 @@ type Provider struct {
 	// provides runtime configuration via the Kubernetes API.
 	//
 	// +optional
-	Kubernetes *KubernetesProvider `json:"kubernetes,omitempty"`
+	Kubernetes *GatewayKubernetesProvider `json:"kubernetes,omitempty"`
```
Comment (arkodg) [inline api/config/v1alpha1/envoygateway_types.go:83]: shoudnt these be updated too with `EnvoyGateway` prefix ?

Comment (qicz) [reply api/config/v1alpha1/envoygateway_types.go]: did it @arkodg

File: api/config/v1alpha1/envoygateway_types.go [hunk 3]
```diff
@@ -80,23 +80,23 @@ type Provider struct {
...
-	Kubernetes *KubernetesProvider `json:"kubernetes,omitempty"`
+	Kubernetes *EnvoyGatewayKubernetesProvider `json:"kubernetes,omitempty"`
 
 	// File defines the configuration of the File provider. File provides runtime
 	// configuration defined by one or more files. This type is not implemented
 	// until https://github.com/envoyproxy/gateway/issues/1001 is fixed.
 	//
 	// +optional
-	File *FileProvider `json:"file,omitempty"`
+	File *GatewayFileProvider `json:"file,omitempty"`
```
Comment (arkodg) [inline api/config/v1alpha1/envoygateway_types.go:90]: `EnvoyGateway` prefix here as well, to make it all consistent ?

Comment (qicz) [reply api/config/v1alpha1/envoygateway_types.go]: found it ,fixed

**Review by arkodg — APPROVED** [review]
LGTM, thanks

**Review by Xunzhuo — APPROVED** [review]
LGTM

**Pull request merged**